include/xtensor/xexpression.hpp

```diff
@@ -302,6 +318,135 @@ namespace xt
+        using inner_backstrides_type = typename E::inner_backstrides_type;
+
+        using stepper = typename E::stepper;
+        using const_stepper = typename E::const_stepper;
+
+        using storage_iterator = typename E::storage_iterator;
+        using const_storage_iterator = typename E::const_storage_iterator;
+
+
+        static constexpr layout_type static_layout = E::static_layout;
+        static constexpr bool contiguous_layout = static_layout != layout_type::dynamic;
+
+        explicit xshared_expression(std::shared_ptr<E>&& ptr)
+            : m_ptr(std::move(ptr))
+        {
+        }
+
+        template <class... Args>
+        auto operator()(Args... args)
+        {
+            return m_ptr->operator()(args...);
+        }
+
+        FORWARD_METHOD(shape);
+        FORWARD_METHOD(dimension);
+        FORWARD_METHOD(size);
+        FORWARD_METHOD(begin);
+        FORWARD_METHOD(cbegin);
+        FORWARD_METHOD(storage_begin);
+        FORWARD_METHOD(storage_cbegin);
+        FORWARD_METHOD(storage_end);
+        FORWARD_METHOD(storage_cend);
+
+        template <class CE = E, class = std::enable_if_t<has_data_interface<CE>::value, int>>
+        FORWARD_METHOD(strides);
+        template <class CE = E, class = std::enable_if_t<has_data_interface<CE>::value, int>>
+        FORWARD_METHOD(data);
+        template <class CE = E, class = std::enable_if_t<has_data_interface<CE>::value, int>>
+        FORWARD_METHOD(data_offset);
+        template <class CE = E, class = std::enable_if_t<has_data_interface<CE>::value, int>>
+        FORWARD_METHOD(storage);
+
+        template <class S>
+        bool broadcast_shape(S& shape, bool reuse_cache = false) const
+        {
+            return m_ptr->broadcast_shape(shape, reuse_cache);
+        }
+
+        template <class S>
+        bool is_trivial_broadcast(const S& strides) const noexcept
+        {
+            return m_ptr->is_trivial_broadcast(strides);
+        }
+
+        template <class S>
+        auto stepper_begin(const S& shape) noexcept
+            -> decltype(std::declval<E>().stepper_begin(shape))
+        {
+            return m_ptr->stepper_begin(shape);
+        }
+
+        template <class S>
+        auto stepper_end(const S& shape, layout_type l) noexcept
+            -> decltype(std::declval<E>().stepper_end(shape, l))
+        {
+            return m_ptr->stepper_end(shape, l);
+        }
+
+        template <class S>
+        auto stepper_begin(const S& shape) const noexcept
+            -> decltype(std::declval<const E>().stepper_begin(shape))
+        {
+            return static_cast<const E*>(m_ptr.get())->stepper_begin(shape);
+        }
+        template <class S>
+        auto stepper_end(const S& shape, layout_type l) const noexcept
+            -> decltype(std::declval<const E>().stepper_end(shape, l))
+        {
+            return static_cast<const E*>(m_ptr.get())->stepper_end(shape, l);
+        }
+
+        auto use_count()
```

Comment:
const?

Reply:
inline is in this case implicit as it's defined in the class body.